cmd/nerdctl/compose_exec_linux_test.go

```diff
@@ -207,3 +209,85 @@ services:
 	base.ComposeCmdWithHelper(unbuffer, "-f", comp.YAMLFullPath(), "exec", "-t=false", "svc0", "stty").AssertFail()                         // `-i`
 	base.ComposeCmdWithHelper(unbuffer, "-f", comp.YAMLFullPath(), "exec", "-i=false", "-t=false", "svc0", "stty").AssertFail()
 }
+
+func TestComposeExecWithIndex(t *testing.T) {
+	testutil.DockerIncompatible(t)
+	base := testutil.NewBase(t)
+	var dockerComposeYAML = fmt.Sprintf(`
+version: '3.1'
+
+services:
+  svc0:
+    image: %s
+    command: "sleep infinity"
+    deploy:
+      replicas: 3
+`, testutil.CommonImage)
+
+	comp := testutil.NewComposeDir(t, dockerComposeYAML)
+	t.Cleanup(func() {
+		comp.CleanUp()
+	})
+	projectName := comp.ProjectName()
+	t.Logf("projectName=%q", projectName)
+
+	base.ComposeCmd("-f", comp.YAMLFullPath(), "up", "-d", "svc0").AssertOK()
+	t.Cleanup(func() {
+		base.ComposeCmd("-f", comp.YAMLFullPath(), "down", "-v").AssertOK()
+	})
+
+	// try 5 times to ensure that results are stable
+	for i := 0; i < 5; i++ {
+		for _, j := range []string{"1", "2", "3"} {
+			host := fmt.Sprintf("%s-svc0-%s.%s_default", projectName, j, projectName)
+			var (
+				expectIP string
+				realIP   string
+			)
+			cmds := []string{"-f", comp.YAMLFullPath(), "exec", "-i=false", "-t=false", "--index", j, "svc0"}
+			base.ComposeCmd(append(cmds, "cat", "/etc/hosts")...).
+				AssertOutWithFunc(func(stdout string) error {
+					lines := strings.Split(stdout, "\n")
+					for _, line := range lines {
+						if !strings.HasSuffix(line, host) {
+							continue
+						}
+						fields := strings.Fields(line)
+						if len(fields) == 0 {
+							continue
+						}
+						expectIP = fields[0]
+						return nil
+					}
+					return errors.New("fail to get the expected ip address")
+				})
+			base.ComposeCmd(append(cmds, "ip", "addr", "show", "dev", "eth0")...).
```

Comment:
Can we use `hostname -i` ?

Reply:
I tried it. and found another inconsistent behavior. When using `hostname -i` for docker, I got only one IP respect with the index. While nerdctl just give me all the three IPs.
I think this is unrelated to the current issue. So let me keep this unchanged.

Reply:
I think `docker compose` has different DNS behavior from `nerdctl compose`. I think `ip addr` would be better during the test